src/wp-includes/class-wp-theme-json.php

```diff
@@ -1501,6 +1519,45 @@ protected static function get_style_nodes( $theme_json, $selectors = array() ) {
 		return $nodes;
 	}
 
+	/**
+	 * For metadata values that can either be booleans or paths to booleans, gets the value.
+	 *
+	 * ```php
+	 * $data = array(
+	 *   'color' => array(
+	 *     'defaultPalette' => true
+	 *   )
+	 * );
+	 *
+	 * static::get_metadata_boolean( $data, false );
+	 * // => false
+	 *
+	 * static::get_metadata_boolean( $data, array( 'color', 'defaultPalette' ) );
+	 * // => true
+	 * ```
+	 *
+	 * @since 6.0.0
+	 *
+	 * @param array      $data The data to inspect.
+	 * @param bool|array $path Boolean or path to a boolean.
+	 * @param bool       $default Default value if the referenced path is missing.
+	 * @return boolean
+	 */
+	protected static function get_metadata_boolean( $data, $path, $default = false ) {
+		if ( is_bool( $path ) ) {
+			return $path;
+		}
+
+		if ( is_array( $path ) ) {
+			$value = _wp_array_get( $data, $path );
+			if ( isset( $value ) ) {
```

Comment:
```suggestion
			if ( null !== $value ) {
```

`isset` is a bit confusing here because you'll never get a PHP warning when accesing the `$value` variable which is a primary purpose of using `isset`. It also matches the style in `::get_data`